climpred/smoothing.py

```diff
@@ -0,0 +1,248 @@
+                Shape can be 1D (Nlon,) and (Nlat,) for rectilinear grids,
+                or 2D (Ny, Nx) for general curvilinear grids.
+                Shape of bounds should be (N+1,) or (Ny+1, Nx+1).
+            d_lon (float): Longitude step size, i.e. grid resolution
+            d_lat (float): Latitude step size, i.e. grid resolution
+        Returns:
+            da : xarray DataArray with coordinate values
+        """
+
+        def check_lon_lat_present(da):
+            if 'lat' in ds.coords and 'lon' in ds.coords:
+                return da
+            elif 'lat_b' in ds.coords and 'lon_b' in ds.coords:
+                return da
+            # for CESM POP grid
+            elif 'TLAT' in ds.coords and 'TLONG' in ds.coords:
+                da = da.rename({'TLAT': 'lat', 'TLONG': 'lon'})
+                return da
+            else:
+                raise ValueError(
+                    'lon/lat or lon_b/lat_b or TLAT/TLON not found, please rename.'
+                )
+
+        da = check_lon_lat_present(da)
+        grid_out = {
+            'lon': np.arange(da.lon.min(), da.lon.max() + d_lon, d_lon),
+            'lat': np.arange(da.lat.min(), da.lat.max() + d_lat, d_lat),
+        }
+        regridder = xe.Regridder(da, grid_out, **kwargs)
+        return regridder(da)
+
+    if 'lon' not in d_lon_lat_dict:
+        d_lon_lat_dict['lon'] = d_lon_lat_dict['lat']
+    elif 'lat' not in d_lon_lat_dict:
+        d_lon_lat_dict['lat'] = d_lon_lat_dict['lon']
+    else:
+        raise ValueError('please provide either `lon` or `lat` in d_lon_lat_dict.')
+
+    kwargs = {
+        'd_lon': d_lon_lat_dict['lon'],
+        'd_lat': d_lon_lat_dict['lat'],
+        'method': method,
+        'periodic': periodic,
+        'filename': filename,
+        'reuse_weights': reuse_weights,
+    }
+
+    ds = _regrid_it(ds, **kwargs)
+
+    return ds
+
+
+def spatial_smoothing_xrcoarsen(ds, coarsen_dict=None, how='mean'):
+    """Apply spatial smoothing by regridding to `boxsize` grid.
+
+    Reference:
+      * Goddard, L., A. Kumar, A. Solomon, D. Smith, G. Boer, P.
+            Gonzalez, V. Kharin, et al. “A Verification Framework for
+            Interannual-to-Decadal Predictions Experiments.” Climate
+            Dynamics 40, no. 1–2 (January 1, 2013): 245–72.
+            https://doi.org/10/f4jjvf.
+
+    Args:
+        ds (xr.object): input. xr.DataArray prefered.
+        coarsen_dict (dict): coarsen spatial latitudes.
+        how (str): aggregation type for coarsening. default: 'mean'
+
+    Returns:
+        ds_smoothed (xr.object): boxsize-regridded input
+
+    """
+    if coarsen_dict is None:
+        # guess spatial dims
+        spatial_dims_to_smooth = list(ds.dims)
+        for dim in ['time', 'lead', 'member', 'init']:
+            if dim in ds.dims:
+                spatial_dims_to_smooth.remove(dim)
+        # write coarsen to dict to coarsen similar to 5x5 degree
+        coarsen_dict = dict()
+        step = 2
```

Comment:
`coarsen` just aggregates `n` grid cells, not latitude/longitude. and `n` needs to divide the dims evenly. saver with `2` as default. this `coarsen` function is a nice try but once we implement xr.methods it will be redundant anyways